src/Raven.Server/Utils/CpuUsageExtensionPoint.cs

```diff
@@ -0,0 +1,263 @@
+                RedirectStandardError = true,
+                CreateNoWindow = true
+            };
+
+            InitNewProcessHandle();
+        }
+
+        public void Start()
+        {
+            try
+            {
+                _process.Start();
+            }
+            catch (Exception e)
+            {
+                NotifyWarning("Could not start cpu usage extension point process", e);
+
+                Dispose();
+            }
+
+            var _ = ReadProcess(); // explicitly starting async task without waiting for it
+            _ = ReadErrors();
+        }
+
+        private async Task ReadErrors()
+        {
+            var errors = await _process.StandardError.ReadLineAsync();
+            if (errors == null)
+            {
+                return;
+            }
+            try
+            {
+                if (_process.HasExited == false)
+                {
+                    _process.Kill();
+                }
+            }
+            catch (Exception)
+            {
+                // When the process terminating, killed or exited
+            }
+            errors = errors + await _process.StandardError.ReadLineAsync();
+            NotifyWarning($"Extension point process send an error: {errors}");
+        }
+
+        public void Dispose()
+        {
+            IsDisposed = true;
+            _data = BadData;
+            using (_process)
+            {
+                try
+                {
+                    if (_process.HasExited == false)
+                    {
+                        _process.Kill();
+                    }
+                }
+                catch (Exception)
+                {
+                    // When the process terminating, killed or exited
+                }
+
+                _process = null;
+            }
+        }
+
+
+        private async Task ReadProcess()
+        {
+            while (IsDisposed == false)
+            {
+                var nextLine = _process.StandardOutput.ReadLineAsync();
+
+                if (await nextLine.WaitWithTimeout(_timeout) == false)
+                {
+                    // here we missed an update, restart the process once
+                    HandleError($"The process didn't send information for {_timeout.TotalSeconds} seconds.");
+                    break;
+                }
+
+                var line = await nextLine;
+                if (HandleInfoReceived(line) == false)
+                {
+                    break;
+                }
+            }
+        }
+
+        private bool HandleInfoReceived(string data)
+        {
+            if (data == null)
+            {
+                HandleError("The output stream of the process has closed.");
+                return false;
+            }
+            using (_contextPool.AllocateOperationContext(out var context))
+            {
+                try
+                {
+                    var blittable = context.ReadForMemory(data, string.Empty);
```

Comment:
Is JSON a way to go when it comes to scripts? It is not a commonly used format.

I would expect output like:
```
80.06 3.0
```

Reply:
We expect this to be a full fledged program, not just a script.
And we want to make sure that we are future proof here.